impala/sqlalchemy.py

```diff
@@ -176,11 +176,28 @@ def get_columns(self, connection, table_name, schema=None, **kwargs):
         cursor.fetchall()
         column_info = []
         for col in schema:
-            column_info.append({
-                'name': col[0],
-                'type': _impala_type_to_sqlalchemy_type[col[1]],
-                'nullable': True,
-                'autoincrement': False})
+            try:
+                column_info.append({
+                    'name': col[0],
+                    'type': _impala_type_to_sqlalchemy_type[col[1]],
+                    'nullable': True,
+                    'autoincrement': False})
+            except KeyError as e:
+                if col[1] == 'VARCHAR':
+                    column_info.append({
+                        'name': col[0],
+                        'type': _impala_type_to_sqlalchemy_type['STRING'],
+                        'nullable': True,
+                        'autoincrement': False})
+                elif col[1] == 'DATE':
+                    col[1] = 'TIMESTAMP'
+                    column_info.append({
+                        'name': col[0],
+                        'type': _impala_type_to_sqlalchemy_type['TIMESTAMP'],
```

Comment:
I think this should map to a Sqlalchemy date type.

Reply:
Totally agree.